Classes/common/CDTDatastore.m

```diff
@@ -396,6 +396,25 @@ - (BOOL)validateBodyDictionary:(NSDictionary *)body error:(NSError *__autoreleas
             return NO;
         }
     }
+    
+    //now to check if the body contains valid json data
+    if(![NSJSONSerialization isValidJSONObject:body]){
```

Comment:
OK guess it depends which you think is the most important error if it is both invalid json and contains keys with underscores.